src/schema/notifications.ts

```diff
@@ -431,4 +442,28 @@ export class ZimbraNotifications {
 			});
 		}
 	};
+
+	private writeToCacheForNewMail = (items: any) => {
+		const itemsToWrite: any = [];
+		items &&
+			items.forEach((i: any) => {
+				const item = normalizeMessage(i);
+				const flags = item.flag || item.flags;
+				flags &&
+					flags.indexOf('u') > -1 &&
+					itemsToWrite.push({
+						id: item.id,
+						subject: item.subject,
+						flags: item.flags || item.flag,
+						folderId: item.folderId,
+						__typename: 'NewMail'
+					});
+			});
+		this.cache.writeQuery({
```

Comment:
we should check if itemsToWrite contains any item or not

Reply:
Added a check at https://github.com/Zimbra/zm-api-js-client/pull/304/files#diff-50cec4d28ea90d367f48b5a273ce67aaR422 so that the function itself won't be called if no items are present.